File: Logic_HandsOn/C++/Problems on Digits,String,Array,Pointers etc/program15.cpp
#include<iostream>
using namespace std;

class Display
{
	private:
		int iValue;

	public:

		Display(int iValue)
		{
			this->iValue = iValue;
		}	

		int CountTwo()
		{
			int iCnt=0;
			while(iValue!=0)
			{
				int iDigit = iValue%10;
				if(iDigit<6)
				{
					iCnt++;
				}
				iValue = iValue/10;
			}

			return iCnt;
		}
};

int main()
{
	int iValue=0,iRet=0;

	cout<<"Enter Number: ";
	cin>>iValue;

	Display *dObj = new Display(iValue);
	iRet = dObj->CountTwo();

	cout<<"Frequency of Digit less than 6 are:  "<<iRet<<"\n";
	return 0;
}

/*
OUTPUT:

Enter Number: 2395
Frequency of Digit less than 6 are:  3


Enter Number: 96672
Frequency of Digit less than 6 are:  1

*/

File: Logic_HandsOn/Java/Problems on Array,Pattern etc/Program15.java
import java.util.*;

class Solution
{
	public int iValue;

	public Solution(int iValue)
	{
		this.iValue = iValue;
	}

	public int Display()
	{
                      	
		int iCnt=0;
		while(iValue!=0)
		{
			int iDigit = iValue%10;
			if(iDigit<6)
			{
				iCnt++;
			}
			iValue = iValue/10;
		}        
		return iCnt;
	
	}
}

class Program5
{
	public static void main(String[] args)
	{
		Scanner sObj = new Scanner(System.in);

		int iRet=0;
		System.out.println("Enter number: ");
		int iNo = sObj.nextInt();

		Solution obj = new Solution(iNo);
		iRet = obj.Display();
	
		System.out.println("Frequency of number less than 6 are: "+iRet);
	}
}

/*
OUTPUT:

Enter number: 
9440
Frequency of number less than 6 are: 3

*/
